src/java/io/bazel/rulesscala/scalac/ScalaCInvoker.java

```diff
@@ -0,0 +1,366 @@
+// Copyright 2014 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package io.bazel.rulesscala.scalac;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.devtools.build.lib.worker.WorkerProtocol.Input;
+import com.google.devtools.build.lib.worker.WorkerProtocol.WorkRequest;
+import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse;
+import io.bazel.rulesscala.jar.JarCreator;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.lang.reflect.Field;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.UUID;
+import java.util.jar.Attributes;
+import java.util.jar.JarOutputStream;
+import java.util.jar.Manifest;
+import scala.Console$;
+import scala.tools.nsc.*;
+import scala.tools.nsc.reporters.ConsoleReporter;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * A class for creating Jar files. Allows normalization of Jar entries by setting their timestamp to
+ * the DOS epoch. All Jar entries are sorted alphabetically.
+ */
+public class ScalaCInvoker {
+  // A UUID that uniquely identifies this running worker process.
+  static final UUID workerUuid = UUID.randomUUID();
+
+  // A counter that increases with each work unit processed.
+  static int workUnitCounter = 1;
+
+  // If true, returns corrupt responses instead of correct protobufs.
+  static boolean poisoned = false;
+
+  // Keep state across multiple builds.
+  static final LinkedHashMap<String, String> inputs = new LinkedHashMap<>();
+
+  static class WorkerOptions {
+    public int exitAfter = 30;
```

Comment:
uhh good question. I lifted most of these things from the example, but not sure if i lifted this value.. I'm just going to kill the poison/exitAfter code